src/compound/lens/RatesLens.sol

```diff
@@ -320,48 +378,108 @@ abstract contract RatesLens is UsersLens {
             uint256 poolBorrowRate
         )
     {
-        ICToken cToken = ICToken(_poolToken);
+        PoolRatesVars memory ratesVars;
+
+        ratesVars.delta = morpho.deltas(_poolToken);
+        ratesVars.lastPoolIndexes = morpho.lastPoolIndexes(_poolToken);
+
+        bool updated = _suppliedOnPool > 0 ||
+            _borrowedFromPool > 0 ||
+            _repaidOnPool > 0 ||
+            _withdrawnFromPool > 0;
+        if (updated) {
+            PoolInterestsVars memory interestsVars;
+            (
+                ratesVars.indexes.poolSupplyIndex,
+                ratesVars.indexes.poolBorrowIndex,
+                interestsVars
+            ) = _accruePoolInterests(ICToken(_poolToken));
+
+            interestsVars.cash =
+                interestsVars.cash +
+                _suppliedOnPool +
+                _repaidOnPool -
+                _borrowedFromPool -
+                _withdrawnFromPool;
+            interestsVars.totalBorrows =
+                interestsVars.totalBorrows +
+                _borrowedFromPool -
+                _repaidOnPool;
+
+            IInterestRateModel interestRateModel = ICToken(_poolToken).interestRateModel();
+
+            poolSupplyRate = IInterestRateModel(interestRateModel).getSupplyRate(
+                interestsVars.cash,
+                interestsVars.totalBorrows,
+                interestsVars.totalReserves,
+                interestsVars.reserveFactorMantissa
+            );
+
+            (bool success, bytes memory result) = address(interestRateModel).staticcall(
+                abi.encodeWithSelector(
+                    IInterestRateModel.getBorrowRate.selector,
+                    interestsVars.cash,
+                    interestsVars.totalBorrows,
+                    interestsVars.totalReserves
+                )
+            );
```

Comment:
I not passing the interface?

Reply:
Because there are 2 types of IRM with 2 types of return values
cUSDC & cETH return 2 `uint256` while other cTokens return 1 `uint256`
This is because all the cTokens don't use the same IRM, so I had to work around to be able to include the 2 IRM interfaces

Reply:
Ok I get it so you can close this conversation